assigner/commands/commit.py

```diff
@@ -82,7 +84,16 @@ def _push(conf, backend, args):
 
             if has_changes or allow_empty:
                 logging.debug("%s: committing changes with message %s", full_name, message)
-                index.commit(message)
+                if gpg_sign:
+                    # The GitPython interface does not support signed commits, and
+                    # launching via repo.git.commit will launch an inaccessible
+                    # interactive prompt in the background
+                    index.write()
+                    subprocess.call(["git", "commit", "-S", "-m", '"{}"'.format(message)],
+                                    stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL,
+                                    cwd=repo_dir)
```

Comment:
We should set `check=True` here as well, that way if `git commit` fails, it won't fail silently. I don't expect that to be a common issue, so no further error handling is required for now. If it turns out to be a problem, we can deal with that in the future.